Feature/161418753 add experimental variables and number of assays

lingyun1010 opened the pull request:
This branch added more experiment information including `numberOfAssays` and `ExperimentalFactors`, which are used in [scxa-faceted-search-results](https://github.com/ebi-gene-expression-group/scxa-faceted-search-results/pull/8).

lingyun1010 committed:
Add number of assays into gene search result json

lingyun1010 committed:
Simplify and refactor code

lingyun1010 committed:
Add a test for ExperimentInfoService

lingyun1010 committed:
Implement a test for ExperimentInfoService

monicajianu reviewed (changes requested):
I think adding a separate service for this is overkill. You already have a service designed to provide information about an experiment, i.e. the `ExperimentsAttributesService`. You should make the number of assays and experimental variables available there.

lingyun1010 committed:
Delete the orignal service and test files

lingyun1010 committed:
Add function for number of cells  and …

experimental factors etc json playload in ExperimentAttributesService and used in JsonGeneSearchController

lingyun1010 committed:
Amend tests for service and controller

lingyun1010 committed:
Add submodules

lingyun1010 committed:
Add a new experiment card component with an icon

lingyun1010 committed:
Specify the type of Map value as ExperimentInfo

lingyun1010 committed:
Make test for the added parameters in json payload

lingyun1010 committed:
Replace ExperimentCard with ExperimentTableCard

lingyun1010 committed:
Add a new package react-ebi-species for icon

lingyun1010 committed:
Update the gene-search bundle

lingyun1010 committed:
Debug the code by correcting the prototype

lingyun1010 committed:
Delete the old experiment card component

lingyun1010 committed:
Modify the new experiment card style

lingyun1010 committed:
Update the JS bundle

Commit:
Simplified retrieval of assay numbers and experimental factors by adding them to the existing method in ExperimentAttributesService. Removed addition of unnecessary duplicate attributes to the search results JSON payload.

Commit:
Changed ExperimentTableCard to use the restructured JSON response. Fixed styling issues caused by unnecessary margins, which caused elements to not be centred inside the div. Replaced x character with EBI cross icon. Made font size consistent across the element (default 16px as for the entire application).

Commit:
Fixed indentation.

Commit:
Removed unused imports.

Commit:
Merge branch 'dev' into feature/161418753-add-experimental-variables-and-number-of-assays

Commit:
Added keys to elements. Fixed indentation.

Commit:
Replaced invalid nested a tags with an onClick handler on the outer div.

Commit:
Rebuilt bundles.

alfonsomunozpomer committed:
Merge branch 'dev' into feature/161418753-add-experimental-variables-and-number-of-assays

# Conflicts:
#	sc/src/main/webapp/resources/js-bundles/geneSearch.bundle.js

alfonsomunozpomer committed:
Replace Foundation tooltips with React Tooltip

alfonsomunozpomer committed:
Remove superfluous SpeciesIconCard

alfonsomunozpomer committed:
Add a provisional header that will be rendered by the results component

alfonsomunozpomer committed:
Update dependencies and generate production bundles

alfonsomunozpomer reviewed (approved):
I’ve added a final few improvements and have everything ready to add a header on top of the results.
